### ISSUE TITLE
Diagnostic suppressor seems to work only for open files

### ISSUE THREAD

**Issue by lennartb-** [dotnet/roslyn#75291]
Hi,

this is somewhat of a follow-up to https://github.com/dotnet/roslyn/issues/68885. I've initially reported this to https://github.com/nunit/nunit.analyzers/issues/786 as well, but it doesn't seem to be an issue with the analyzer itself.

For context: The NUnit.Analyzer package provides diagnostic suppressor to suppress nullable warnings for objects that can be null, but have previously been checked with NUnit's asserts that they are not null.

We recently changed a legacy test project that previously had NRTs disabled to enable them. We saw hundreds of "CS8602 - Dereference of a possibly null reference" warnings for objects that can be in fact null, but were checked with some variation of "Assert this is not null". Curiously, the warnings weren't reported at build time and disappeared after the file with the warning was opened.

Here's a repro, mostly based on the Roslyn issue above:
Project file:
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
    <IsPackable>false</IsPackable>
    <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
    <Nullable>enable</Nullable>
    <ImplicitUsings>enable</ImplicitUsings>
  </PropertyGroup>

  <PropertyGroup>
    <AccelerateBuildsInVisualStudio>true</AccelerateBuildsInVisualStudio>
    <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.11.1" />
    <PackageReference Include="NUnit" Version="4.2.2">
      <NoWarn>NU1608</NoWarn>
    </PackageReference>
    <PackageReference Include="NUnit.Analyzers" Version="4.3.0">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
  </ItemGroup>
</Project>
```

Test:
```csharp
using NUnit.Framework;

namespace NUnitNullabilityTest
{
    [TestFixture]
    public class Tests
    {
        private class Model
        {
            public string? Name { get; set; }

            public string? SomeMethod() => Name;
        }

        [Test]
        public void Model_SomeMethod()
        {
            var target = new Model { Name = "name" };

            var result = target.SomeMethod();

            Assert.That(result, Is.Not.Null);
            Assert.That(result.Length, Is.EqualTo(4)); // This is of course a nonsense assertion, but repros the issue
        }
    }
}

```

`result` can technically be null, so `result.Length` could theoretically throw an NRE. Since `Is.Not.Null` is asserted, we can be sure it's not null.

This produces the warning if the file is not opened:
![image](https://github.com/user-attachments/assets/68a9f255-5d14-4ea8-b078-d2e055b158be)

An vanishes if the file is opened:
![image](https://github.com/user-attachments/assets/3841e942-b98d-4fed-9c27-c98045242dfc)

**Comment by datvm** [comment]
Somebody on [StackOverflow](https://stackoverflow.com/a/78441374/653457) reports an issue with EF Core as well.

This seems to be an UI bug only as it only shows up in VS, `dotnet build` does not actually report any warning and Error List with `Build Only` (instead of `Build + IntelliSense`) also does not show anything.

**Comment by lennartb-** [comment]
There is also a VS feedback ticket that seems to be related: https://developercommunity.visualstudio.com/t/Private-Unity-messages-incorrectly-marke/10779025

**Comment by lennartb-** [comment]
@arunchndr This is not fixed as of VS 17.12.3 and still repros.

**Comment by arunchndr** [comment]
@lennartb- the diagnostic suppressors you linked above did ship in 17.12.3. Is your original repro steps still intact on 17.12.3?

**Comment by arunchndr** [comment]
Reopening.

**Comment by lennartb-** [comment]
@arunchndr Yes, the repro from the original post is still intact:

![Image](https://github.com/user-attachments/assets/fdc8edd6-00f9-42a4-9640-e7a8490fe898)
![Image](https://github.com/user-attachments/assets/48805df9-4920-4b32-829c-5985556a1b70)
